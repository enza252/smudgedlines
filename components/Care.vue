<template>
  <v-container>
    <v-row class="ml-2">
      <h1 id="care-header">Care</h1>
    </v-row>
    <v-row>
      <template v-for="(card, i) in cards">
        <v-col :key="i">
          <v-card color="accent" rounded elevation="10" min-height="100%">
            <v-card-title style="justify-content: center">
              {{ card.title }}
              <v-icon v-if="card.icon" right >{{ card.icon }}</v-icon>
            </v-card-title>
            <v-card-text style="text-align: center">
              {{ card.content }}
            </v-card-text>
          </v-card>
        </v-col>
      </template>
    </v-row>
  </v-container>
</template>
<script>
import { defineComponent } from '@nuxtjs/composition-api'
import { mdiWashingMachine, mdiRadiator } from '@mdi/js'
export default defineComponent({
  name: 'Care',
  setup () {

  },
  data () {
    return {
      cards: [
        {
          title: 'Washing',
          content:
            'Machine Wash at 30 degrees celsius on a low spin, or hand wash in a bowl of warm water.',
          icon: mdiWashingMachine
        },
        {
          title: 'Drying',
          content: 'Air dry our Tees on a washing line or near a warm radiator. Do not tumble dry - it\'s bad for the planet and will shrink our Tees!',
          icon: mdiRadiator
        }
      ]
    }
  }
})
</script>
