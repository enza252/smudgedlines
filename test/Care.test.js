import Vuetify from 'vuetify'
import { screen, render } from '@testing-library/vue'
import Care from '@/components/Care'

const options = {
  routes: [],
  vuetify: new Vuetify()
}

describe('tests the care component renders correctly', () => {
  it('tests the component against the snapshot', async () => {
    const { asFragment } = await render(Care, options)
    expect(asFragment).toMatchSnapshot()
  })

  it('tests the care component contains the expected instructions', async () => {
    await render(Care, options)
    expect(screen.getByText('Care')).toBeInTheDocument()
    expect(screen.getByText('Machine Wash at 30 degrees celsius on a low spin, or hand wash in a bowl of warm water.')).toBeInTheDocument()
    expect(screen.getByText('Air dry our Tees on a washing line or near a warm radiator. Do not tumble dry - it\'s bad for the planet and will shrink our Tees!')).toBeInTheDocument()
  })

})
