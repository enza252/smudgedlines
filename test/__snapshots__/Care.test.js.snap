// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the care component renders correctly tests the component against the snapshot 1`] = `undefined`;
